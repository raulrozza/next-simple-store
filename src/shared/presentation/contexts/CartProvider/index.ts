export { default as useAddToCart } from './hooks/useAddToCart';
