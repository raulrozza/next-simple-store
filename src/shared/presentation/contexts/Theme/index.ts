export { default as useTheme } from './hooks/useTheme';
export { default as Theme } from './implementations/StyledComponentTheme';
