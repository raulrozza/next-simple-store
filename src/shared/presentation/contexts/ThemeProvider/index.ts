export { default as useTheme } from './hooks/useTheme';
export { default as ThemeProvider } from './implementations/StyledComponentTheme';
