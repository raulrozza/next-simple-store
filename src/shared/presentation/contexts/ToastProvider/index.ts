export { useToastProvider } from './context';
export { default as ToastProvider } from './implementations/ReactToastifyProvider';
