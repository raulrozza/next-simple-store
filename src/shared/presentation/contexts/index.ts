export * from './CartProvider';
export * from './ThemeProvider';
