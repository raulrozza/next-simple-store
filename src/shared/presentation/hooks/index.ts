import { trpc } from '@/shared/infra/http/trpc';

const { useQuery, useMutation } = trpc;

export { useQuery, useMutation };
