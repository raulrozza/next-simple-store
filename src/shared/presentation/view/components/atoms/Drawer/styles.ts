import styled, { css } from 'styled-components';

export const Check = styled.input`
    display: none;
`;

interface HiddenProps {
    $hidden: boolean;
}

export const Container = styled.nav<HiddenProps>`
    position: fixed;
    visibility: hidden;
    background: white;
    transition: all;
    box-shadow: 0 0 10px 5px #00000029;
    z-index: 101;
    transition-duration: 500ms;
    top: 0;
    left: 0;
    transform: translate3d(-100%, 0, 0);
    width: 200px;
    height: 100vh;

    ${({ $hidden }) =>
        !$hidden &&
        css`
            visibility: visible;
            transform: translate3d(0, 0, 0) !important;
        `}
`;

export const Overlay = styled.label<HiddenProps>`
    display: none;
    height: 100vh;
    left: 0;
    position: fixed;
    top: 0;
    width: 100%;
    background-color: #000;
    opacity: 0.4;
    z-index: 100;

    ${({ $hidden }) =>
        !$hidden &&
        css`
            display: block;
        `}
`;
