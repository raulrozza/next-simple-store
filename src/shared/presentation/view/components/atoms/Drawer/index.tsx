import { FC } from 'react';

import { Check, Container, Overlay } from './styles';

type DrawerProps = {
    open: boolean;
    onClose: () => void;
};

const Drawer: FC<DrawerProps> = ({ open, onClose, children, ...props }) => (
    <div {...props}>
        <Check
            type="checkbox"
            id="drawer-checkbox"
            onChange={onClose}
            checked={open}
        />

        <Container role="navigation" $hidden={!open}>
            {children}
        </Container>

        <Overlay
            htmlFor="drawer-checkbox"
            $hidden={!open}
            data-testid="drawer-Overlay"
        />
    </div>
);

export default Drawer;
