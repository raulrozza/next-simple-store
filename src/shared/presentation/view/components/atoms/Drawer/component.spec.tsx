import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { ThemeProvider } from '@/shared/presentation/contexts';

import Drawer from './index';

describe('Drawer', () => {
    it('should render the component hiding the content', () => {
        const view = render(
            <Drawer open={false} onClose={jest.fn()}>
                Content
            </Drawer>,
            {
                wrapper: ThemeProvider,
            },
        );

        expect(screen.queryByRole('navigation')).toBeNull();
        expect(screen.getByTestId('drawer-Overlay')).toHaveStyle({
            display: 'none',
        });
        expect(view).toMatchSnapshot();
    });

    it('should render the component showing the content if the open prop is true', () => {
        const view = render(
            <Drawer open={true} onClose={jest.fn()}>
                Content
            </Drawer>,
            {
                wrapper: ThemeProvider,
            },
        );

        expect(screen.getByRole('navigation')).toHaveStyle({
            visibility: 'visible',
        });
        expect(screen.getByTestId('drawer-Overlay')).toHaveStyle({
            display: 'block',
        });
        expect(view).toMatchSnapshot();
    });

    it('should show call the onClose callback when the overlay is clicked', async () => {
        const onClose = jest.fn();
        const user = userEvent.setup();

        render(
            <Drawer open={true} onClose={onClose}>
                Content
            </Drawer>,
            {
                wrapper: ThemeProvider,
            },
        );

        await user.click(screen.getByTestId('drawer-Overlay'));

        expect(onClose).toHaveBeenCalled();
    });
});
