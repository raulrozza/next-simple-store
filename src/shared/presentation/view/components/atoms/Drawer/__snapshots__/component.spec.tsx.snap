// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer should render the component hiding the content 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: none;
}

.c1 {
  position: fixed;
  visibility: hidden;
  background: white;
  -webkit-transition: all;
  transition: all;
  box-shadow: 0 0 10px 5px #00000029;
  z-index: 101;
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  top: 0;
  left: 0;
  -webkit-transform: translate3d(-100%,0,0);
  -ms-transform: translate3d(-100%,0,0);
  transform: translate3d(-100%,0,0);
  width: 200px;
  height: 100vh;
}

.c2 {
  display: none;
  height: 100vh;
  left: 0;
  position: fixed;
  top: 0;
  width: 100%;
  background-color: #000;
  opacity: 0.4;
  z-index: 100;
}

<body>
    <div>
      <div>
        <input
          class="c0"
          id="drawer-checkbox"
          type="checkbox"
        />
        <nav
          class="c1"
          role="navigation"
        >
          Content
        </nav>
        <label
          class="c2"
          data-testid="drawer-Overlay"
          for="drawer-checkbox"
        />
      </div>
    </div>
  </body>,
  "container": .c0 {
  display: none;
}

.c1 {
  position: fixed;
  visibility: hidden;
  background: white;
  -webkit-transition: all;
  transition: all;
  box-shadow: 0 0 10px 5px #00000029;
  z-index: 101;
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  top: 0;
  left: 0;
  -webkit-transform: translate3d(-100%,0,0);
  -ms-transform: translate3d(-100%,0,0);
  transform: translate3d(-100%,0,0);
  width: 200px;
  height: 100vh;
}

.c2 {
  display: none;
  height: 100vh;
  left: 0;
  position: fixed;
  top: 0;
  width: 100%;
  background-color: #000;
  opacity: 0.4;
  z-index: 100;
}

<div>
    <div>
      <input
        class="c0"
        id="drawer-checkbox"
        type="checkbox"
      />
      <nav
        class="c1"
        role="navigation"
      >
        Content
      </nav>
      <label
        class="c2"
        data-testid="drawer-Overlay"
        for="drawer-checkbox"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Drawer should render the component showing the content if the open prop is true 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: none;
}

.c1 {
  position: fixed;
  visibility: hidden;
  background: white;
  -webkit-transition: all;
  transition: all;
  box-shadow: 0 0 10px 5px #00000029;
  z-index: 101;
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  top: 0;
  left: 0;
  -webkit-transform: translate3d(-100%,0,0);
  -ms-transform: translate3d(-100%,0,0);
  transform: translate3d(-100%,0,0);
  width: 200px;
  height: 100vh;
  visibility: visible;
  -webkit-transform: translate3d(0,0,0) !important;
  -ms-transform: translate3d(0,0,0) !important;
  transform: translate3d(0,0,0) !important;
}

.c2 {
  display: none;
  height: 100vh;
  left: 0;
  position: fixed;
  top: 0;
  width: 100%;
  background-color: #000;
  opacity: 0.4;
  z-index: 100;
  display: block;
}

<body>
    <div>
      <div>
        <input
          checked=""
          class="c0"
          id="drawer-checkbox"
          type="checkbox"
        />
        <nav
          class="c1"
          role="navigation"
        >
          Content
        </nav>
        <label
          class="c2"
          data-testid="drawer-Overlay"
          for="drawer-checkbox"
        />
      </div>
    </div>
  </body>,
  "container": .c0 {
  display: none;
}

.c1 {
  position: fixed;
  visibility: hidden;
  background: white;
  -webkit-transition: all;
  transition: all;
  box-shadow: 0 0 10px 5px #00000029;
  z-index: 101;
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  top: 0;
  left: 0;
  -webkit-transform: translate3d(-100%,0,0);
  -ms-transform: translate3d(-100%,0,0);
  transform: translate3d(-100%,0,0);
  width: 200px;
  height: 100vh;
  visibility: visible;
  -webkit-transform: translate3d(0,0,0) !important;
  -ms-transform: translate3d(0,0,0) !important;
  transform: translate3d(0,0,0) !important;
}

.c2 {
  display: none;
  height: 100vh;
  left: 0;
  position: fixed;
  top: 0;
  width: 100%;
  background-color: #000;
  opacity: 0.4;
  z-index: 100;
  display: block;
}

<div>
    <div>
      <input
        checked=""
        class="c0"
        id="drawer-checkbox"
        type="checkbox"
      />
      <nav
        class="c1"
        role="navigation"
      >
        Content
      </nav>
      <label
        class="c2"
        data-testid="drawer-Overlay"
        for="drawer-checkbox"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
